backend/src/connector/mod.rs

```diff
@@ -1 +1,309 @@
+mod prelude {
+    pub use super::api::prelude::*;
+    pub use super::{
+        context::{ContextWrapper, DropContextService, Has},
+        ClientError, Connector,
+    };
+    pub use crate::{models::shared::XSpanIdString, prelude::*, services::auth::HttpClient};
+    pub use axum::{
+        headers::{authorization::Credentials, Authorization, HeaderMapExt},
+        http::{HeaderName, HeaderValue},
+    };
+    pub use futures::StreamExt;
+    pub use hyper::{service::Service, Response, Uri};
+    pub use std::{
+        str::FromStr,
+        sync::Arc,
+        task::{Context, Poll},
+    };
+}
 
+use api::{ApiNoContext, ContextWrapperExt};
+use client::Client;
+use context::ClientContext;
+use prelude::*;
+
+use self::error::AsApiError;
+
+pub mod api;
+pub mod client;
+pub mod context;
+pub mod dto;
+pub mod error;
+
+pub type ApiInterface = dyn ApiNoContext<ClientContext> + Send + Sync;
+
+#[derive(Debug, Error)]
+pub enum ClientError {
+    #[error("couldn't init http client")]
+    InitClient,
+    #[error("couldn't probe connection to the node")]
+    Inaccessible,
+    #[error("permission denied")]
+    PermissionDenied,
+    #[error("no client found for requested resource")]
+    NoClient,
+    #[error("can't form hyper request")]
+    CantFormRequest,
+    #[error("bad URI")]
+    BadUri,
+}
+
+/// HTTP Connector constructor
+#[derive(Debug)]
+pub struct Connector;
+
+impl Connector {
+    /// Alows building a HTTP(S) connector. Used for instantiating clients with custom
+    /// connectors.
+    #[must_use]
+    pub const fn builder() -> Builder {
+        Builder {}
+    }
+}
+
+/// Builder for HTTP(S) connectors
+#[derive(Debug)]
+pub struct Builder {}
+
+impl Builder {
+    /// [Stub] Use HTTPS instead of HTTP
+    #[must_use]
+    pub const fn https(self) -> HttpsBuilder {
+        HttpsBuilder {}
+    }
+
+    /// Build a HTTP connector
+    #[must_use]
+    pub fn build(self) -> hyper::client::connect::HttpConnector {
+        hyper::client::connect::HttpConnector::new()
+    }
+}
+
+// TODO
+/// [Stub] Builder for HTTPS connectors
+#[derive(Debug)]
+pub struct HttpsBuilder {}
+
+impl HttpsBuilder {
+    pub fn build(self) {
+        unimplemented!()
+    }
+}
+
+#[derive(Clone)]
+pub struct BobClient<Client: ApiNoContext<ClientContext> + Send + Sync> {
```

Comment:
Perhaps, it is better to call it `BobClusterClient`, because inside it contains clients for all nodes